src/lib/pipeline.ml

```diff
@@ -777,4 +792,21 @@ module Compiler = struct
     in
     compiler.wrap_gtf_node pipeline gtf_node
 
-end
+  let hla_types_step ~compiler (pipeline : hla_types pipeline) =
```

Comment:
These methods are also all in the `Compiler` module; `Compiler.compile_hla_types_step` seems redundant. Other methods (`fastq_step` and `fastq_sample_step`) were also factored out into separate functions.